pkg/tbtc/tbtc.go

```diff
@@ -53,10 +54,18 @@ func Initialize(
 	persistence persistence.Handle,
 	scheduler *generator.Scheduler,
 	config Config,
+	registry *diagnostics.Registry,
 ) error {
 	node := newNode(chain, netProvider, persistence, scheduler, config)
 	deduplicator := newDeduplicator()
 
+	assembleTbtcDiagnostics := func() map[string]interface{} {
+		return map[string]interface{}{
+			"preParamsPoolSize": node.dkgExecutor.PreParamsPool().CurrentSize(),
```

Comment:
I am not sure if we should expose `PreParamsPool` on the DKG executor. The pre-params pool feels like something really internal to DKG and if we have the executor abstraction, it should hide the complexity.

This comment also applies to #3273 which depends on this PR. 

Although it's a bit more of boilerplate code, I would prefer to do it in the following way:

```
func assembleDiagnostics() map[string]interface{}  {
  dkgDiagnostics := node.dkgExecutor.Diagnostics()

  diagnostics := make(map[string]interface{})
  for k, v := range dkgDiagnostics {
    diagnostics[k] = v
  }

  return diagnostics
}
```

And then here, call just `registry.RegistryApplicationSource("tbtc", assembleDiagnostics)`.

We may find ourselves in a situation when we'd want to add diagnostics for signing as well. Every sub-package of `tbtc` would be responsible for assembling its own internal diagnostics.

I do not want to block this PR on it, but maybe that's something we can address in #3273 @lukasz-zimnoch ?

Reply:
See my response at https://github.com/keep-network/keep-core/pull/3273#discussion_r970831193.

However, regarding diagnostics specifically, I have mixed feelings about "Every sub-package of tbtc would be responsible for assembling its own internal diagnostics.". The concept of diagnostics is something _above_ the application and something the application should not be aware of. I think each application component (like DKG executor, tbtc node, etc) should just expose some basic information about their current state and that is. The layer above should gather this and expose it as diagnostics or metrics.

Reply:
The reasoning about the application component gathering the metrics is solid but I still think that in the current setup we have an abstraction leak. Pre-params pool is something that is used by the executor and is created in `dkg` package. Only inside `dkg` package we should call `GetNow()` on the pool.

With `node.dkgExecutor.PreParamsPool()` exposes from the executor, the pool is leaking and `GetNow()` can be called from `tecdsa`.

If we do not want each sub-package to fill diagnostics - which after second thought I do agree could be overkill - I would expose `CurrentPreParamsPoolSize()` from the executor instead of returning the entire pool.

Reply:
Yep, this is what I proposed in https://github.com/keep-network/keep-core/pull/3273#discussion_r970831193 as well.

Reply:
I now see @lukasz-zimnoch proposed the same [here](https://github.com/keep-network/keep-core/pull/3273#discussion_r970831193) so I think we should do this change in #3273.